src/endpoint/egress.cpp

```diff
@@ -324,126 +251,36 @@ void endpoint::Egress::manage_exiting_tcp_packets(unsigned char* pkt,
-
-    //exit(EXIT_FAILURE);
-
-    /*send(socket, s, pkt_len - pkt_calc, 0);
-
-    LOG(ldebug, "3");
-    //ssize_t received_len = 0;
-    unsigned char buffer[BUFFER_SIZE];
-    //char* sfcid;
-    //char* next_ip;
-    //uint16_t next_port;
-    //unsigned long ttl;
-
-    do {
-        LOG(ldebug, "4");
-        received_len = read(socket, buffer, BUFFER_SIZE);
-        printf("received ");
-        for(int i = 0; i < received_len; i++)
-            printf("%x", buffer);
-        printf("\n");
-        LOG(ldebug, "5");
-        if (received_len < 0) {
-            perror("error receiving data");
-            exit(EXIT_FAILURE);
-        } else if (received_len > 0) {
-            LOG(ldebug, "6");
-            sfcid = classifier_.classify_pkt(buffer, received_len);
-            LOG(ldebug, "7");
-            std::vector<db::utils::Address> path =
-                    roulette_->get_route_list(atoi(sfcid));
-            LOG(ldebug, "8");
-
-            if (!path.empty()) {
-                // +2 because of ingress & egress
-                ttl = path.size() + 2;
-                next_ip = const_cast<char *>(path[0].get_address().c_str());
-                next_port = path[0].get_port();
-                LOG(ldebug, "9");
-                sfc_header flh =
-                        utils::sfc_header::SFCUtilities::create_header(
-                                atoi(sfcid), 0,
-                                INT_TO_IP_C_STR(header.source_address),
-                                header.source_port,
-                                INT_TO_IP_C_STR(header.destination_address),
-                                header.destination_port, ttl, 1);
-                unsigned char formatted_pkt[pkt_len + SFC_HDR];
-                unsigned char* pkt_pointer = formatted_pkt;
-                LOG(ldebug, "10");
-                utils::sfc_header::SFCUtilities::prepend_header(buffer,
-                                                                pkt_len,
-                                                                flh,
-                                                                pkt_pointer);
-                LOG(ldebug, "11");
-                LOG(ldebug, "next ip   " + std::string(next_ip));
-                LOG(ldebug, "next port " + std::to_string(next_port));
-                client::udp::ClientUDP().send_and_wait_response(pkt_pointer,
-                                                                pkt_len + SFC_HDR,
-                                                                next_ip,
-                                                                next_port);
-
-
-
-            } else {
-                LOG(ldebug, "no route available, discarding packages");
-            }
-        }
-
-        if (received_len < BUFFER_SIZE) {
-            break;
-        }
-
-    }  while(1);*/
-
-    //free(pkt);
-    //free(buffer);
+    } while (true);
 
     LOG(ldebug, "exit managing TCP exiting pkts");
 
     // TODO think when closing socket
```

Comment:
It's this TODO done?

Reply:
More or less, connection cleanup is not treated correctly

Reply:
Are you going to fix it this pr or do you think we can safely merge without this resolved for now?

Reply:
I think it is better to manage all claenup in a new PR

Reply:
ok